File: src/org/jetbrains/agda/core/CApplication.java
package org.jetbrains.agda.core;

/**
 * @author Evgeny.Kurbatsky
 */
public class CApplication extends CExpression {
    private final CExpression myLeft;
    private final CExpression myRight;

    public CApplication(CExpression left, CExpression right) {
        this.myLeft = left;
        this.myRight = right;
        attach(left);
        attach(right);
    }

    public CExpression getLeft() {
        return myLeft;
    }

    public CExpression getRight() {
        return myRight;
    }

    @Override
    public String toString() {
        return "(" + myLeft + " " + myRight + ")";
    }
}

File: src/org/jetbrains/agda/core/expression/CApplication.kt
package org.jetbrains.agda.core

/**
 * @author Evgeny.Kurbatsky
 */
public class CApplication(val left : CExpression, val right : CExpression) : CExpression() {
    {
        attach(left)
        attach(right)
    }
    fun toString() : String {
        return "(" + left + " " + right + ")"
    }
    fun equals(other : Any) : Boolean {
        if (other is CApplication) {
            return other.left == left && other.right == right;
        }
        return false;
    }
}
